rfcs/2020-04-23-2445-log-schema.md

```diff
@@ -0,0 +1,240 @@
+   `log_schema.aws_ec2_key` to `"cloud"`.
+2. We should favor namespacing this data since it'll help to avoid breaking
+   changes as this matures. For example, instead of `socket` we should have
+   a `socket.address` key.
+3. We should try to avoid bikeshedding event names as much as possible. The
+   options should correlate 1 to 1 with every field we touch in Vector. We can
+   default the names to whatever we want.
+
+## Prior Art
+
+1. [Elastic Common Schema][ecs]
+2. [Timber Schema][timber_schema]
+3. [OpenTelemtry log data model proposal][ot_log_data_model]
+
+## Sales Pitch
+
+1. The Vector log schema controls will be discoverable and editable in one
+   place.
+2. It creates an explicit contract across Vector.
+3. It increases Vector's adoption rate. The ability to easily adopt an existing
+   log schema is critical to acquiring larger users with legacy systems. Log
+   schemas are arguably the biggest source of lock-in when coming from other
+   tools.
+4. We can be flexible and adopt new standards as they emerge. For example,
+   OpenTelemetry is [working on a schema standard][ot_log_data_model] that I
+   would like Vector to _optionally_ adopt when ready.
+
+## Drawbacks
+
+1. It's more complicated.
+2. I'm unsure if this will affect performance negatively. Adding more fields
+   usually means we're paying a cost.
+
+## Rationale & Alternatives
+
+1. Do nothing and let user rename fields on their own via `rename_fields`, etc.
+2. Adopt the [ECS][ecs] schema directly and not bother adding options to
+   customize it. This is exactly how Logstash currently works.
+3. Add fine-grained options within every component, allowing users to change
+   fields on a component level. For example, a user can change `host_key` in
+   the `socket` source and also change it in the `datadog_logs` sink to ensure
+   both components are operating on the same field.
+
+## Outstanding Questions
+
+None right now.
+
+## Plan of Attack
+
+### `log_schema` options
+
+- Preset
+  - [ ] Add a new `log_schema.preset` option, this is used to preset all of these values (ex: "v1" and "ecs").
+- Root
+  - Keep the `log_schema.host_key` option (default `host`).
+  - Keep the `log_schema.message_key` option (default `message`).
+  - [ ] Add a new `log_schema.stream_key` option (default `stream`).
+  - Keep the `log_schema.timestamp_key` option (default `timestamp`).
+- [ ] Add a new `log_schema.containers_key` option (default `containers`).
+  - [ ] Add a new `log_schema.containers.created_at_key` option (default `created_at`).
+  - [ ] Add a new `log_schema.containers.id_key` option (default `id`).
+  - [ ] Add a new `log_schema.containers.name_key` option (default `name`).
+  - [ ] Add a new `log_schema.containers.image_key` option (default `image`).
+  - [ ] Add a new `log_schema.containers.labels_key` option (default `labels`).
+- [ ] Add a new `log_schema.file_key` option (default `file`).
+  - [ ] Add a new `log_schema.file.path_key` option (default `path`).
```

Comment:
This doesn't seem to namespace sources/sinks/transforms and may lead to awkward collisions later -- as well as a painful migration. Suggest namespacing fully.